### PR TITLE
RTEMIS-201: Added students bulk import for student tracking.

### PR DESCRIPTION
**Pull request by manish-3108**
Ticket # : https://innoraft.atlassian.net/browse/RTEMIS-201

---
### What the PR does
`(Explain fixes, changes and impacts)`
- Added file upload form to import students.
- Added batch process to import studnets.

---
### What I have tested
`(Explain usecases you have tested locally)`
- Tested that the feature works as expected.

---

### CODE REVIEW

File: modules/rte_mis_core/src/Controller/LogsDownload.php [hunk 1]
```diff
@@ -148,4 +148,83 @@ public function getLocationLogs(int $fid = 0) {
     return new Response('File not found!', Response::HTTP_NOT_FOUND);
   }
 
+  /**
+   * Export the logs from student performance import.
+   */
```
Comment (kunal-sin) [inline modules/rte_mis_core/src/Controller/LogsDownload.php:153]: Function param comment missing.

File: modules/rte_mis_core/src/Controller/LogsDownload.php [hunk 2]
```diff
@@ -148,4 +148,83 @@ public function getLocationLogs(int $fid = 0) {
...
+        $spreadsheet = IOFactory::load($newFileUri);
+        // Make changes to the spreadsheet.
+        $sheet = $spreadsheet->getActiveSheet();
+        // Get the store collection.
+        $store = $this->tempStoreFactory->get('rte_mis_student_tracking');
+        $studentPerformanceLogs = $store->get('student_performance_logs');
+        if (!empty($studentPerformanceLogs)) {
+          $sheet->getStyle('A1:R1')->applyFromArray([
+            'font' => [
+              'bold' => TRUE,
```
Comment (kunal-sin) [inline modules/rte_mis_core/src/Controller/LogsDownload.php:174]: Looks like this code is getting repeated.

File: modules/rte_mis_student_tracking/src/Controller/StudentTrackingLogsDownload.php [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        $newFileUri = $this->fileSystem->realpath($newFile->getFileUri());
+        $spreadsheet = IOFactory::load($newFileUri);
+        // Make changes to the spreadsheet.
+        $sheet = $spreadsheet->getActiveSheet();
+        // Get the store collection.
+        $store = $this->tempStoreFactory->get('rte_mis_student_tracking');
+        $studentsImportLogs = $store->get('students_import_logs');
+        if (!empty($studentsImportLogs)) {
+          $sheet->getStyle('A1:M1')->applyFromArray([
+            'font' => [
```
Comment (kunal-sin) [inline modules/rte_mis_student_tracking/src/Controller/StudentTrackingLogsDownload.php:94]: Looks like this code is getting repeated. Can't we create a helper function for this?

File: modules/rte_mis_core/src/Controller/LogsDownload.php (hunk 2 above)
Comment (manish-3108) [reply modules/rte_mis_core/src/Controller/LogsDownload.php:174]: Right removing this.

File: modules/rte_mis_core/src/Controller/LogsDownload.php (hunk 1 above)
Comment (manish-3108) [reply modules/rte_mis_core/src/Controller/LogsDownload.php]: Resolved

File: modules/rte_mis_student_tracking/src/Controller/StudentTrackingLogsDownload.php (hunk 3 above)
Comment (manish-3108) [reply modules/rte_mis_student_tracking/src/Controller/StudentTrackingLogsDownload.php:100]: This is needed only once I pushed duplicate code by mistake.

**Review by kunal-sin — APPROVED** [review]

**Pull request merged**